pcapgo/read.go

```diff
@@ -101,6 +101,10 @@ func (r *Reader) ReadPacketData() (data []byte, ci gopacket.CaptureInfo, err err
 	}
 
 	var n int
+	if 16+ci.CaptureLength > len(r.buf) {
+		err = gopacket.ErrMalformedPacket
```

Comment:
For now, maybe just a regular error with more context:

err = fmt.Errorf("capture length with header exceeds buffer size: %d > %d", 16+ci.CaptureLength, len(r.buf))

If you're dead set on having an error up in packet.go, please add a comment to it:

// ErrMalformedPacket is returned when ...
var ErrMalformedPacket ...